sql/_13_issues/_23_1h/cases/cbrd_24843_2.sql

```diff
@@ -0,0 +1,279 @@
+/* ----------------------------------------
+ *
+ * test case 36:
+ *
+ * 1. 'c.p_a = 1' -> pt_lambda_node() -> 'p.a = 1' [o]
+ * 2. 'p.a = 2' -> pt_lambda_node() -> 'c.p_a = 2' [o]
+ * 3. 'p.a = 3 and p.b = 3' -> pt_lambda_node() -> 'c.p_a = 3 and c.p_b = 3' [o]
+ * 4. 'p.a > 3 and p.a = 4' -> pt_lambda_node() -> 'c.p_a = 4' [o]
+ * 5. '(c.p_a = 5 or c.p_b = 5)' [o]
+ * 6. '(p.a = 6 or p.b = 6)' [x]
+ * 7. 'p.a = cast (7 as numeric (1, 0))' [o]
+ * 8. 'p.a >= 8' [x]
+ * 9. 'p.a in (1, 3, 5, 7, 9)' [x]
+ * 10. 'p.a between 2 and 4' [x]
+ * 11. 'p.a = (select a from t_parent order by a desc limit 1)' [x]
+ *
```

Comment:
12. 'c.p_a <> 1' ->
    'c.p_a != 1' ->
13. 'c.p_a LIKE 1' -> 

14. select c.b, count(*), min(c.p_a), max(c.p_b) 
    from t_child as c, t_parent as p
    where c.p_a = p.a and c.p_b = p.b
      and p.a = 4
    group by c.b

15. select c.b, count(*), min(c.p_a), max(c.p_b) 
    from t_child as c, t_parent as p
    where c.p_a = p.a and c.p_b = p.b
      and p.a = 4
    group by c.b
    having count(*) > 2

16. select c.*, count(*) over ( PARTITION BY c.b ORDER BY c.a DESC) AS cnt
    from t_child as c, t_parent as p
    where c.p_a = p.a and c.p_b = p.b

17. select c.*, count(*) over ( PARTITION BY p.b ORDER BY p.a DESC) AS cnt
    from t_child as c, t_parent as p
    where c.p_a = p.a and c.p_b = p.b

18. select c.*, count(*) over ( PARTITION BY c.b ORDER BY p.a DESC) AS cnt
    from t_child as c, t_parent as p
    where c.p_a = p.a and c.p_b = p.b

19. EXISTS (select a from t_parent where a = c.p_a and b = c.p_b )'

20. NOT EXISTS (select a from t_parent where a = c.p_a and b = c.p_b )'

21. select /*+ recompile */
           c.*,
           ( select cc.a from t_child cc where cc.p_a=p.a and cc.p_b=p.b limit 1 ) chk_scala_x
    from
         t_child as c,
         t_parent as p
    where
        c.p_a = p.a and c.p_b = p.b
    and c.p_a = 1;

22. select /*+ recompile */
           c.*,
           ( select pp.a from t_parent pp where pp.a=c.a and pp.b=c.b limit 1 ) chk_scala_o
    from
         t_child as c,
         t_parent as p
    where
        c.p_a = p.a and c.p_b = p.b
    and c.p_a = 1;

23. select /*+ recompile */
           c.*, level
    from
         t_child as c,
         t_parent as p
    where
        c.p_a = p.a and c.p_b = p.b
    connect by prior c.a = c.b
    
24. WITH
 of_ooooo AS (
    select /*+ recompile */ 'ooooo', c.*
    from t_child as c, t_parent as p
    where c.p_a = p.a and c.p_b = p.b
    and c.p_a = 1 ),
 of_xxxxx AS (
    select /*+ recompile */ 'xxxxx', c.*
    from t_child as c, t_parent as p
    where c.p_a = p.a and c.p_b = p.b
    and p.a >= 8
     )
SELECT * FROM of_ooooo 
UNION ALL 
SELECT * FROM of_xxxxx 
ORDER BY 1;

Reply:
suggested cases are added.